lab_1_keywords_tfidf/main.py

```diff
@@ -4,106 +4,138 @@
-
     Parameters:
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str):
+        return None
+    for punc in string.punctuation:
+        text = text.replace(punc, '')
+        clean_words = text.lower().strip().split()
+    return clean_words
 
 
 def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
     Excludes stop words from the token sequence
-
     Parameters:
     tokens (List[str]): Original token sequence
     stop_words (List[str]: Tokens to exclude
-
     Returns:
     List[str]: Token sequence that does not include stop words
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(isinstance(tokens, list) and tokens != [] and all(isinstance(token, str) for token in tokens)
+            and isinstance(stop_words, list)):
+        return None
+    no_stop_words = [token for token in tokens if token not in stop_words]
+    return no_stop_words
 
 
 def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
     """
     Composes a frequency dictionary from the token sequence
-
     Parameters:
     tokens (List[str]): Token sequence to count frequencies for
-
     Returns:
     Dict: {token: number of occurrences in the token sequence} dictionary
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(isinstance(tokens, list) and tokens != []
+            and all(isinstance(token, str) for token in tokens)):
+        return None
+    freq_dict = {}
+    for token in tokens:
+        if token not in freq_dict:
+            freq_dict[token] = 1
+        else:
+            freq_dict[token] += 1
+    return freq_dict
 
 
 def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[list[str]]:
     """
     Extracts a certain number of most frequent tokens
-
     Parameters:
     frequencies (Dict): A dictionary with tokens and
     its corresponding frequency values
     top (int): Number of token to extract
-
     Returns:
     List[str]: Sequence of specified length
     consisting of tokens with the largest frequency
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(isinstance(frequencies, dict) and frequencies != {}
+            # and all(isinstance(token, str) for word in frequencies.keys())
+            # and all(isinstance(freq, int or float) for freq in frequencies.values())
+            and isinstance(top, int) and top is not (True or False) and top > 0):
+        return None
+
+    sorted_freq_dict = {token: token for token, freq
+                        in sorted(frequencies.items(), key=lambda token: token[1], reverse=True)}
+    sorted_words = list(sorted_freq_dict.keys())
+    top_words = sorted_words if top > len(frequencies) else sorted_words[:top]
```

Comment:
is it neccessary to check this condition? would `top_words = sorted_words[:top]` work just as fine in both cases?